### PR TITLE
Publishing latency SLI

### PR DESCRIPTION
**Pull request by mike29736**
https://trello.com/c/jDSDGNyn/814-measure-and-record-our-publishing-latency-sli

This is a Ruby script. It's designed at least a bit with extension in mind. This pull request introduces a single SLI object and some minimal supporting code to make that runnable from a cronjob.

The SLI object performs these steps:
* Fetches Whitehall events from Logit
* Fetches Content Store events from Logit
* Turns those events into metrics
* Sends those metrics to Prometheus's PushGateway

I hope the commits are self-explanatory.

Things I'm unsure about:
- how to get stdout logs into Logit
- how best to get exceptions into Sentry (I'm assuming the `sentry-ruby` gem rather than the `govuk_app_config` gem for `GovukError`)
- what a Dockerfile for a project like this should look like (is this a good example to follow? https://github.com/alphagov/govuk-dependency-checker/blob/main/Dockerfile)

### CODE REVIEW

File: lib/govuk_sli_collector/publishing_latency_sli/logit_search.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+require "rest-client"
```
Comment (mike29736) [inline lib/govuk_sli_collector/publishing_latency_sli/logit_search.rb:1]: It turns out that I forgot to `require "json"` here, so I'm just going to force-push a fix for that in a second :adhesive_bandage:

**Review by JonathanHallam — COMMENTED** [review]
I'm super happy with the changes you've made but don't have answers to the questions you've put in the PR description. Is this something oyu want to pair on before you merge or shall I approve now for you?

**Comment by mike29736** [comment]
> I'm super happy with the changes you've made but don't have answers to the questions you've put in the PR description. Is this something oyu want to pair on before you merge or shall I approve now for you?

Thanks, @JonathanHallam. I'm happy for this to go into `main` before those problems are resolved.

(I've opened another pull request with the first pass at a Dockerfile, BTW. I'll start looking into logging and exceptions now.)

**Review by JonathanHallam — APPROVED** [review]
Thanks!

**Pull request merged**